src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,53 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final String REPORT = "Report for period ";
+    private static final String DASH = " - ";
+    private static final String SEPARATOR = " ";
+    private static final int HOURLY_RATE_SECOND = 1;
+    private static final int HOURLY_RATE_THIRD = 2;
+    private static final int HOURLY_RATE_FOURTH = 3;
+    private static final int HOURLY_RATE_ZERO = 0;
+    private static final DateTimeFormatter DATE_TIME_FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder report = new StringBuilder();
+        LocalDate isBefore = LocalDate.parse(dateFrom, DATE_TIME_FORMATTER);
+        LocalDate isAfter = LocalDate.parse(dateTo, DATE_TIME_FORMATTER);
+        int totalSalary = 0;
+
+        report.append(REPORT).append(dateFrom)
+                .append(DASH).append(dateTo);
+
+        for (String name : names) {
+            int employeeSalary = calculateEmployeeSalary(name,data,isBefore,isAfter);
```

Comment:
a space after each comma
use IDE's autoformatter before committing - Ctrl Alt L for windows machines